arch/arm64/core/cortex_r/arm_mpu.c

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (c) 2017 Linaro Limited.
+ * Copyright (c) 2021 Arm Limited (or its affiliates). All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <init.h>
+#include <kernel.h>
+#include <soc.h>
+#include <arch/arm64/mm.h>
+#include <linker/linker-defs.h>
+
+#define LOG_LEVEL CONFIG_MPU_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mpu);
+
+#define MPU_DYNAMIC_REGION_AREAS_NUM	1
+
+#define _MAX_DYNAMIC_MPU_REGIONS_NUM                                                               \
+	((IS_ENABLED(CONFIG_USERSPACE) ? (CONFIG_MAX_DOMAIN_PARTITIONS + 1) : 0) +                 \
+	 (IS_ENABLED(CONFIG_MPU_STACK_GUARD) ? 1 : 0))
+
+#ifdef CONFIG_USERSPACE
+static int dynamic_areas_init(uintptr_t start, size_t size);
+#define _MPU_DYNAMIC_REGIONS_AREA_START ((uintptr_t)&_app_smem_start)
```

Comment:
Nit: Why are these macros prefixed with an underscore?

Reply:
I thought it is only for the inside of this file. Anyway, I'll remove the underscore